packages/resource-deployment/scripts/process-utilities.sh

```diff
@@ -12,20 +12,69 @@ function waitForProcesses() {
     for pid in "${!list}"; do
         echo "Waiting for process with pid $pid"
         wait $pid
-        echo "Process with pid $pid exited"
+        local pExitCode=$?
+        echo "Process with pid $pid exited with exit code $pExitCode"
+
+        if [[ $pExitCode != 0 ]]; then
+            echo "Process - $pid failed with exit code $pExitCode. Killing current proccess."
+            exit $pExitCode
```

Comment:
where do we kill process in this func?

Reply:
not here. it is used in install_paralle.

Reply:
still message is confusing here `Killing current proccess.` also typo here

Reply:
i can change the message if u prefer. but, i don;t see typo.

Reply:
yes